From d3ec32193c00ef599e9f240342433d18cbe403c5 Mon Sep 17 00:00:00 2001
From: Vladimir Lomov <vladimir@theor7.ktp>
Date: Tue, 2 Mar 2021 09:48:02 +0800
Subject: [PATCH 21/30] scripts/gen-work.py: added script to generate

  - json "workload" file. Currently we generate two files: one with
    "direct" order of tasks and another with shuffled tasks.

Signed-off-by: Vladimir Lomov <vladimir@theor7.ktp>
---
 scripts/gen-work.py | 34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)
 create mode 100755 scripts/gen-work.py

diff --git a/scripts/gen-work.py b/scripts/gen-work.py
new file mode 100755
index 0000000..2b75e6b
--- /dev/null
+++ b/scripts/gen-work.py
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
+
+import json
+import os
+from pathlib import Path
+import random
+
+methods = [ "M2", "M4", "M6", "CF4" , "CF4:3" ]
+steps = [ 1e-3, 1e-4, 1e-5, 1e-6, 1e-7, 1e-8, 1e-9, 1e-10 ]
+rngs = [ (0.1, 0.15), (0.1, 0.2), (0.1, 0.3) ]
+
+work = []
+
+json_dir = Path("methods")
+
+## We didn't handle the case when json_dir exists but not a directory.
+if not os.path.isdir(json_dir):
+    os.mkdir(json_dir)
+
+for r in rngs:
+    for m in methods:
+        for s in steps:
+            work.append({
+                "method" : f"{m}",
+                "start" : r[0],
+                "stop" : r[1],
+                "step": s})
+
+shfld = random.sample(work, k=len(work))
+
+with open(json_dir / "sample1.json", "w") as f:
+    json.dump({ "work" : shfld }, f, indent="  ")
+with open(json_dir / "sample2.json", "w") as f:
+    json.dump({ "work" : work }, f, indent="  ")
-- 
2.30.1

